packages/ipx/src/ipx.ts

```diff
@@ -217,6 +207,37 @@ class IPX {
       await this.cache.clean()
     }
   }
+
+  private initInputAdapters (adapters: any) {
+    this.inputs = adapters.map((adapter: any): BaseInputAdapter => {
+      // Create instance of input
+      let InputCtor: { new(ipx: IPX): BaseInputAdapter }
+      if (typeof adapter === 'string') {
+        InputCtor = (InputAdapters as any)[adapter] || require(resolve(adapter))
+      } else {
+        InputCtor = adapter
+      }
+      const input = new InputCtor(this)
+      if (typeof input.init === 'function') {
+        input.init()
+      }
+      return input
+    })
+  }
+
+  private async stats (src: string): Promise<Stats | false> {
+    const input = this.inputs.find(inp => inp.test(src))
+    if (input) {
+      return await input.stats(src)
+    }
+    return false
+  }
+
+  private async get (src: string) {
+    const input = this.inputs.find(inp => inp.test(src))
```

Comment:
We are still moving dispatch logic to inputs which can make conflicts. We can instead split src and check by prefix (Like `/cloudinary/path/to/asset`). It is a breaking change for IPX core but since we generate URLs using client, client will add prefix.

Reply:
Hmm, you are suggesting to implement runtime adapter detection based on request url?  
It will change IPX URL structure to this, right?
`http://domain/[format]/[operation]/[adapter]/[image-path]`

Reply:
Yes. I would suggest `http://domain/[adapter]/[format]/[operation]/[image-path]` to have easier caching control on edge layer (`doman/[adapter]` rules)